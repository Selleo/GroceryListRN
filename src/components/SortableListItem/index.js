import React, { PureComponent } from 'react'
import { Animated, TouchableOpacity, StyleSheet } from 'react-native'
import { Text } from 'react-native-paper'

import { gray, white, transparent } from 'src/styles/colors'

type Props = {
  move: Function,
  moveEnd: Function,
  isActive: boolean,
  item: object,
}

export class SortableListItem extends PureComponent<Props> {
  state = {
    animation: new Animated.Value(0),
  }

  componentDidMount = () => {
    this.loopAnimation()
  }

  loopAnimation = () => {
    this.animation = Animated.loop(
      Animated.timing(this.state.animation, {
        duration: 200,
        isInteraction: false,
        toValue: 1,
      }),
    ).start()
  }

  render() {
    const {
      move,
      moveEnd,
      isActive,
      item: { name },
    } = this.props
    const animationValue = this.state.animation.interpolate({
      inputRange: [0, 0.25, 0.5, 0.75, 1],
      outputRange: ['0deg', '1deg', '0deg', '-1deg', '0deg'],
    })

    return (
      <Animated.View style={{ transform: [{ rotateZ: animationValue }] }}>
        <TouchableOpacity
          onPressIn={move}
          onPressOut={moveEnd}
          style={[styles.container, { backgroundColor: isActive ? gray : white }]}
        >
          <Text numberOfLines={2} style={styles.text}>
            {name}
          </Text>
        </TouchableOpacity>
      </Animated.View>
    )
  }
}

export default SortableListItem

const styles = StyleSheet.create({
  container: {
    flex: 1,
    height: 50,
    margin: 5,
    paddingVertical: 10,
    paddingHorizontal: 20,
    justifyContent: 'space-between',
    flexDirection: 'column',
  },
  text: {
    fontWeight: 'bold',
    backgroundColor: transparent,
  },
})
