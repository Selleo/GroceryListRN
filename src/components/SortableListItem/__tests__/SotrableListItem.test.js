import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import SortableListItem from '../'

describe('<SortableListItem />', () => {
  const defaultProps = {
    editItem: jest.fn(),
    move: jest.fn(),
    moveEnd: jest.fn(),
    isActive: false,
    item: { name: 'ItemName' },
  }

  test('render', () => {
    const wrapper = renderer.create(<SortableListItem {...defaultProps} />)

    expect(wrapper).toMatchSnapshot()
  })

  test('render inActive item', () => {
    const wrapper = renderer.create(<SortableListItem {...defaultProps} isActive={true} />)

    expect(wrapper).toMatchSnapshot()
  })
})
