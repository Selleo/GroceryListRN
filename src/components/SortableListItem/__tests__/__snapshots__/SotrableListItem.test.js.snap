// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortableListItem /> render 1`] = `
<RCTView
  style={
    Object {
      "transform": Array [
        Object {
          "rotateZ": Object {
            "interpolate": [MockFunction] {
              "calls": Array [
                Array [
                  Object {
                    "inputRange": Array [
                      0,
                      0.25,
                      0.5,
                      0.75,
                      1,
                    ],
                    "outputRange": Array [
                      "0deg",
                      "1deg",
                      "0deg",
                      "-1deg",
                      "0deg",
                    ],
                  },
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": [Circular],
                },
              ],
            },
          },
        },
      ],
    }
  }
>
  <RCTView
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
            "height": 50,
            "justifyContent": "space-between",
            "margin": 5,
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          },
          Object {
            "backgroundColor": "#FFFFFF",
          },
        ],
        Object {
          "opacity": Object {
            "__makeNative": [MockFunction],
            "addListener": [MockFunction],
            "interpolate": [MockFunction] {
              "calls": Array [
                Array [
                  Object {
                    "inputRange": Array [
                      0,
                      0.25,
                      0.5,
                      0.75,
                      1,
                    ],
                    "outputRange": Array [
                      "0deg",
                      "1deg",
                      "0deg",
                      "-1deg",
                      "0deg",
                    ],
                  },
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": Object {
                    "interpolate": [MockFunction] {
                      "calls": Array [
                        Array [
                          Object {
                            "inputRange": Array [
                              0,
                              0.25,
                              0.5,
                              0.75,
                              1,
                            ],
                            "outputRange": Array [
                              "0deg",
                              "1deg",
                              "0deg",
                              "-1deg",
                              "0deg",
                            ],
                          },
                        ],
                      ],
                      "results": [Circular],
                    },
                  },
                },
              ],
            },
            "removeAllListeners": [MockFunction],
            "removeListener": [MockFunction],
            "start": [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            },
          },
        },
      ]
    }
  >
    <Text
      numberOfLines={2}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Helvetica Neue",
          },
          Object {
            "backgroundColor": "transparent",
            "fontWeight": "bold",
          },
        ]
      }
      theme={
        Object {
          "colors": Object {
            "accent": "#03dac4",
            "background": "#f6f6f6",
            "disabled": "rgba(0, 0, 0, 0.26)",
            "error": "#B00020",
            "placeholder": "rgba(0, 0, 0, 0.38)",
            "primary": "#6200ee",
            "surface": "#ffffff",
            "text": "#000000",
          },
          "dark": false,
          "fonts": Object {
            "light": "HelveticaNeue-Light",
            "medium": "HelveticaNeue-Medium",
            "regular": "Helvetica Neue",
            "thin": "HelveticaNeue-Thin",
          },
          "roundness": 4,
        }
      }
    >
      ItemName
    </Text>
  </RCTView>
</RCTView>
`;

exports[`<SortableListItem /> render inActive item 1`] = `
<RCTView
  style={
    Object {
      "transform": Array [
        Object {
          "rotateZ": Object {
            "interpolate": [MockFunction] {
              "calls": Array [
                Array [
                  Object {
                    "inputRange": Array [
                      0,
                      0.25,
                      0.5,
                      0.75,
                      1,
                    ],
                    "outputRange": Array [
                      "0deg",
                      "1deg",
                      "0deg",
                      "-1deg",
                      "0deg",
                    ],
                  },
                ],
                Array [
                  Object {
                    "inputRange": Array [
                      0,
                      0.25,
                      0.5,
                      0.75,
                      1,
                    ],
                    "outputRange": Array [
                      "0deg",
                      "1deg",
                      "0deg",
                      "-1deg",
                      "0deg",
                    ],
                  },
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": Object {
                    "interpolate": [MockFunction] {
                      "calls": Array [
                        Array [
                          Object {
                            "inputRange": Array [
                              0,
                              0.25,
                              0.5,
                              0.75,
                              1,
                            ],
                            "outputRange": Array [
                              "0deg",
                              "1deg",
                              "0deg",
                              "-1deg",
                              "0deg",
                            ],
                          },
                        ],
                        Array [
                          Object {
                            "inputRange": Array [
                              0,
                              0.25,
                              0.5,
                              0.75,
                              1,
                            ],
                            "outputRange": Array [
                              "0deg",
                              "1deg",
                              "0deg",
                              "-1deg",
                              "0deg",
                            ],
                          },
                        ],
                      ],
                      "results": [Circular],
                    },
                  },
                },
                Object {
                  "isThrow": false,
                  "value": [Circular],
                },
              ],
            },
          },
        },
      ],
    }
  }
>
  <RCTView
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
            "height": 50,
            "justifyContent": "space-between",
            "margin": 5,
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          },
          Object {
            "backgroundColor": "#AEAEAE",
          },
        ],
        Object {
          "opacity": Object {
            "__makeNative": [MockFunction],
            "addListener": [MockFunction],
            "interpolate": [MockFunction] {
              "calls": Array [
                Array [
                  Object {
                    "inputRange": Array [
                      0,
                      0.25,
                      0.5,
                      0.75,
                      1,
                    ],
                    "outputRange": Array [
                      "0deg",
                      "1deg",
                      "0deg",
                      "-1deg",
                      "0deg",
                    ],
                  },
                ],
                Array [
                  Object {
                    "inputRange": Array [
                      0,
                      0.25,
                      0.5,
                      0.75,
                      1,
                    ],
                    "outputRange": Array [
                      "0deg",
                      "1deg",
                      "0deg",
                      "-1deg",
                      "0deg",
                    ],
                  },
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": Object {
                    "interpolate": [MockFunction] {
                      "calls": Array [
                        Array [
                          Object {
                            "inputRange": Array [
                              0,
                              0.25,
                              0.5,
                              0.75,
                              1,
                            ],
                            "outputRange": Array [
                              "0deg",
                              "1deg",
                              "0deg",
                              "-1deg",
                              "0deg",
                            ],
                          },
                        ],
                        Array [
                          Object {
                            "inputRange": Array [
                              0,
                              0.25,
                              0.5,
                              0.75,
                              1,
                            ],
                            "outputRange": Array [
                              "0deg",
                              "1deg",
                              "0deg",
                              "-1deg",
                              "0deg",
                            ],
                          },
                        ],
                      ],
                      "results": [Circular],
                    },
                  },
                },
                Object {
                  "isThrow": false,
                  "value": Object {
                    "interpolate": [MockFunction] {
                      "calls": Array [
                        Array [
                          Object {
                            "inputRange": Array [
                              0,
                              0.25,
                              0.5,
                              0.75,
                              1,
                            ],
                            "outputRange": Array [
                              "0deg",
                              "1deg",
                              "0deg",
                              "-1deg",
                              "0deg",
                            ],
                          },
                        ],
                        Array [
                          Object {
                            "inputRange": Array [
                              0,
                              0.25,
                              0.5,
                              0.75,
                              1,
                            ],
                            "outputRange": Array [
                              "0deg",
                              "1deg",
                              "0deg",
                              "-1deg",
                              "0deg",
                            ],
                          },
                        ],
                      ],
                      "results": [Circular],
                    },
                  },
                },
              ],
            },
            "removeAllListeners": [MockFunction],
            "removeListener": [MockFunction],
            "start": [MockFunction] {
              "calls": Array [
                Array [],
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            },
          },
        },
      ]
    }
  >
    <Text
      numberOfLines={2}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Helvetica Neue",
          },
          Object {
            "backgroundColor": "transparent",
            "fontWeight": "bold",
          },
        ]
      }
      theme={
        Object {
          "colors": Object {
            "accent": "#03dac4",
            "background": "#f6f6f6",
            "disabled": "rgba(0, 0, 0, 0.26)",
            "error": "#B00020",
            "placeholder": "rgba(0, 0, 0, 0.38)",
            "primary": "#6200ee",
            "surface": "#ffffff",
            "text": "#000000",
          },
          "dark": false,
          "fonts": Object {
            "light": "HelveticaNeue-Light",
            "medium": "HelveticaNeue-Medium",
            "regular": "Helvetica Neue",
            "thin": "HelveticaNeue-Thin",
          },
          "roundness": 4,
        }
      }
    >
      ItemName
    </Text>
  </RCTView>
</RCTView>
`;
